net/portmapper/portmapper.go

```diff
@@ -454,6 +467,35 @@ func (c *Client) createOrGetMapping(ctx context.Context) (external netip.AddrPor
 		return netip.AddrPort{}, NoMappingError{ErrGatewayIPv6}
 	}
 
+	now := time.Now()
+
+	// Log what kind of portmap we obtained
+	portmapType := "none"
```

Comment:
Rather than storing various things in this, can you use `c.mapping.MappingType()` in the log messages?

Reply:
Hm, I think I can refactor that to work. The big thing is that I wanted to distinguish between "a new mapping" (`pcp`) and "reused an existing one" (`existing-pcp`).